### PR TITLE
ENH: Don't require specific order for fNIRS

### PR DESCRIPTION
**Pull request by larsoner**
- [x] Push commit that should be green that does some refactoring to use `zip(picks[::2], picks[1::2])`
- [x] Change `_check_channels_ordered` to return `picks` even if they aren't in matched pairs
- [x] Add tests that operating on reordered channels is the same as original order:
    - [x] optical density
    - [x] SCI
    - [x] TDDR
    - [x] beer_lambert

cc @rob-luke I'll ping you when this is ready to go

### CODE REVIEW

**Comment by larsoner** [comment]
Okay @rob-luke I think this one is ready. I still think we could do some better refactoring of `_check_channels_ordered`, `_channel_frequencies`, `_channel_chromophore`, and `_validate_nirs_info` -- I think eventually ideally unifying within just `_validate_nirs_info` -- but for now this is probably good enough.

**Comment by rob-luke** [comment]
Wow!! That was quick @larsoner

I'm currently in transit to the US and won't be able to review this for a few days. When's the 1.1 release scheduled for? I'd love to review, but will I be holding things up?

**Comment by larsoner** [comment]
Argh I removed a bunch of the `nominal` channel stuff and it broke hitachi... I'll have to fix it.

**Comment by larsoner** [comment]
@rob-luke this one is good to go

**Review by rob-luke — COMMENTED** [review]
This is amazing @larsoner, thanks for tackling this.

Something we will want to support in the future is NIRS data with more than two frequencies of light. This is beyond the scope of this PR. But I wanted to raise it in advance incase it would change any of your design decisions here. I read the code with this in mind and nothing appears to block future extension.

With this PR can we now remove the following documentation?

https://github.com/mne-tools/mne-python/blob/5bddef897c97bdcfb8d879901137c599e776df56/tutorials/io/30_reading_fnirs_data.py#L28-L35

https://github.com/mne-tools/mne-python/blob/5bddef897c97bdcfb8d879901137c599e776df56/tutorials/io/30_reading_fnirs_data.py#L189-L197

File: mne/preprocessing/nirs/nirs.py [hunk 1]
```diff
@@ -181,45 +217,47 @@ def _throw_or_return_empty(msg, throw_errors):
...
-def _validate_nirs_info(info, throw_errors=True):
+def _validate_nirs_info(info, *, throw_errors=True, fnirs=None, which=None,
+                        check_bads=True, allow_empty=True):
     """Apply all checks to fNIRS info. Works on all continuous wave types."""
-    freqs = np.unique(_channel_frequencies(info, nominal=True))
+    _validate_type(fnirs, (None, str), 'fnirs')
+    kinds = dict(
+        od='optical density',
+        cw_amplitude='continuous wave',
+        hbx='chromophore',
```
Comment (rob-luke) [inline mne/preprocessing/nirs/nirs.py:227]:
```suggestion
        hb='chromophore',
```

**Comment by larsoner** [comment]
Not enough sleep this weekend apparently... we already store the frequency, so maybe my next PR should be this automatic pairing based on `chs['loc']` to avoid using channel names completely. Then people can name channels however they want in the future, which would be nice.

**Comment by larsoner** [comment]
In it goes, thanks for the quick review @rob-luke ! There might be some adjustments necessary at the MNE-NIRS end, do you have time to look there?

**Comment by rob-luke** [comment]
> There might be some adjustments necessary at the MNE-NIRS end, do you have time to look there?

I do, will do today.

On another note @larsoner, NIRx have changed the way they store triggers in their latest software version. I have been sent a file with their new format. I will open a PR today for this too, so that we can get this in for 1.1.

**Pull request merged**